packages/imperative/src/censor/src/Censor.ts

```diff
@@ -0,0 +1,340 @@
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import * as lodash from "lodash";
+import { Arguments } from "yargs";
+import { ICommandProfileProperty } from "../../cmd/src/doc/profiles/definition/ICommandProfileProperty";
+import { CliUtils } from "../../utilities/src/CliUtils";
+import { ICensorOptions } from "./doc/ICensorOptions";
+import { Config } from "../../config/src/Config";
+import { ImperativeConfig } from "../../utilities/src/ImperativeConfig";
+import { EnvironmentalVariableSettings } from "../../imperative/src/env/EnvironmentalVariableSettings";
+import { ICommandProfileTypeConfiguration } from "../../cmd/src/doc/profiles/definition/ICommandProfileTypeConfiguration";
+import { IProfileSchema} from "../../profiles/src/doc/definition/IProfileSchema";
+import { IProfileTypeConfiguration } from "../../profiles/src/doc/config/IProfileTypeConfiguration";
+
+export class Censor {
+
+    /*********************************************************************
+    * Basic censorship items - list definitions & initialiazations, etc. *
+    **********************************************************************/
+
+    /*
+    * NOTE(Kelosky): Ideally we might have a consolidated list for secure fields, but for now we'll just
+    * make sure they're collocated within the same class.
+    *
+    * NOTE(Harn): This list should be kept in sync with the base profile secure definitions and MUST be in camel case.
+    */
+    private static readonly MAIN_CENSORED_OPTIONS = ["auth", "authentication", "basicAuth", "certFilePassphrase", "credentials",
+        "pw", "pass", "password", "passphrase", "tv", "tokenValue"];
+
+    private static readonly MAIN_SECURE_PROMPT_OPTIONS = ["keyPassphrase", "password", "passphrase", "tokenValue", "user"];
+
+    // The censor response.
+    public static readonly CENSOR_RESPONSE = "****";
+
+    // A set of default censored options.
+    public static get DEFAULT_CENSORED_OPTIONS(): string[] {
+        const censoredList = new Set<string>();
+        for (const option of this.MAIN_CENSORED_OPTIONS) {
+            censoredList.add(option);
+            censoredList.add(CliUtils.getOptionFormat(option).kebabCase);
+        }
+        return Array.from(censoredList);
+    }
+
+    // Return a customized list of secure prompt options
+    public static get SECURE_PROMPT_OPTIONS(): string[] {
+        const censoredList = new Set<string>();
+        for (const option of this.MAIN_SECURE_PROMPT_OPTIONS) {
+            censoredList.add(option);
+            censoredList.add(CliUtils.getOptionFormat(option).kebabCase);
+        }
+        return Array.from(censoredList);
+    }
+
+    // Set a censored options list that can be set and retrieved for each command.
+    private static censored_options: Set<string> = new Set(this.DEFAULT_CENSORED_OPTIONS);
```

Comment:
Not anything major, but I noticed that this property is formatted as lowercase & snake case. Shortly after in this class, the `m` prefix and camel case is used to declare the remaining private class members. Please update this to follow the same naming scheme